app/jinja_filters.py

```diff
@@ -34,30 +40,30 @@ def strip_tags(value: str) -> Markup:
 
 
 @blueprint.app_template_filter()
-def format_number(value: Union[int, Decimal, float]) -> str:
-    formatted_number: str
-    if value or value == 0:
-        formatted_number = numbers.format_decimal(
-            value, locale=flask_babel.get_locale()
-        )
-        return formatted_number
+def format_number(value: int | Decimal | float) -> str:
+    locale = flask_babel.get_locale()
 
-    return ""
+    formatted_number: str = custom_format_decimal(value, locale)
+    return formatted_number
 
 
 def get_formatted_address(address_fields: dict[str, str]) -> str:
     address_fields.pop("uprn", None)
     return "<br>".join(address_field for address_field in address_fields.values())
 
 
-def get_formatted_currency(value: Union[float, Decimal], currency: str = "GBP") -> str:
-    if value or value == 0:
-        formatted_currency: str = numbers.format_currency(
-            number=value, currency=currency, locale=flask_babel.get_locale()
-        )
-        return formatted_currency
-
-    return ""
+def get_formatted_currency(
+    value: int | float | Decimal,
+    currency: str = "GBP",
+    decimal_limit: int | None = None,
```

Comment:
We should be consistent with the naming for clarity, some are called limit, some are called places.

Reply:
This awkward because was trying to differentiate between number of decimal places and the decimal limit set in the schema, but in some instances we use the number of decimal places as the limit

Reply:
I would name it relevant to the usage. If it is used to work out the number of places, then it should be called limit, if it is used as the literal places, then use places.